### ISSUE TITLE
WARN logs when using SentinelServers config

### ISSUE THREAD

**Issue by ramprasad1996** [redisson/redisson#4003]
**Expected behavior**
Connected using SentinelServer config, all the Redis pods(1master, 2 slave) are up and running in the Kubernetes cluster. Sentinels are also running along with the Redis containers like a sidecar. No WARN logs to be printed when connection is fine.
I am connecting to the Redis headless service in kube.

**Actual behavior**
Getting continuous WARN logs like below for all the IPs where Redis pods are running

02 Dec 2021 14:03:15,355  WARN [redisson-netty-5-30]  SentinelConnectionManager:546 - sentinel: redis://10.244.164.253:26379 is down
 02 Dec 2021 14:03:15,356  WARN [redisson-netty-5-30]  SentinelConnectionManager:546 - sentinel: redis://10.244.154.103:26379 is down
 02 Dec 2021 14:03:15,356  WARN [redisson-netty-5-30]  SentinelConnectionManager:546 - sentinel: redis://10.244.226.111:26379 is down
 02 Dec 2021 14:03:19,548  WARN [redisson-netty-2-19]  SentinelConnectionManager:546 - sentinel: redis://10.244.164.253:26379 is down
 02 Dec 2021 14:03:19,548  WARN [redisson-netty-2-19]  SentinelConnectionManager:546 - sentinel: redis://10.101.141.35:26379 is down
 02 Dec 2021 14:03:19,548  WARN [redisson-netty-2-19]  SentinelConnectionManager:546 - sentinel: redis://10.244.226.111:26379 is down
 02 Dec 2021 14:03:19,552  INFO [redisson-netty-2-22]  SentinelConnectionManager:613 - sentinel: redis://10.102.102.69:26379 added
 02 Dec 2021 14:03:19,731  INFO [redisson-netty-5-25]  SentinelConnectionManager:613 - sentinel: redis://10.244.164.253:26379 added
 02 Dec 2021 14:03:19,732  INFO [redisson-netty-5-21]  SentinelConnectionManager:613 - sentinel: redis://10.244.154.103:26379 added
 02 Dec 2021 14:03:19,732  INFO [redisson-netty-5-23]  SentinelConnectionManager:613 - sentinel: redis://10.244.226.111:26379 added
 02 Dec 2021 14:03:19,885  INFO [redisson-netty-2-26]  SentinelConnectionManager:613 - sentinel: redis://10.244.226.111:26379 added

**Steps to reproduce or test case**
3 Redis pods running along with Sentinels 
App connected to Redis with Sentinel config

**Redis version**
6.2.3

**Redisson version**
3.16.5

**Redisson configuration**
 String redisUrls = "redis://redis:26379";
        config.useSentinelServers()
        .setReadMode(ReadMode.MASTER_SLAVE)
        .setScanInterval(5000)
        .setRetryAttempts(2)
        .setRetryInterval(1000)
        .setPassword(redisPassword)
        .setPingConnectionInterval(5000)
        .setMasterName("redis-master")
        .addSentinelAddress(redisUrls.split(","));

Please let me know if i have mis-configured anything. Thanks.

**Comment by mrniko** [comment]
This may happen then sentinels report different nodes. Since Redisson request them in round-robin.

**Comment by gkormaris** [comment]
Hi @mrniko ,

I face the same issue. From your question above I did not understand whether this is just "noise" in application logs, which could be avoided by reducing log level, or a real issue?

I deploy Redis using bitnami/helm chart in Openshift.

**Redis Version**
Bitnami/Redis helm chart version 15.5.5
1 Master with two Replicas and sentinels enabled.

**Redisson version**
3.16.5

**Log Excerpt**
```
2021-12-07 16:03:47.506  WARN [ocp-permissions,,,] 1 --- [sson-netty-2-16] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is down
2021-12-07 16:03:48.509  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-7] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is up
2021-12-07 16:03:48.512  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-7] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:03:48.513  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-12] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:03:58.524  WARN [ocp-permissions,,,] 1 --- [isson-netty-2-9] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is down
2021-12-07 16:03:59.526  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-2] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is up
2021-12-07 16:03:59.528  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-4] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:03:59.530  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-8] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:04:00.528  WARN [ocp-permissions,,,] 1 --- [sson-netty-2-11] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is down
2021-12-07 16:04:01.529  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-3] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is up
2021-12-07 16:04:01.531  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-15] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:04:01.532  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-10] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:05:04.631  WARN [ocp-permissions,,,] 1 --- [sson-netty-2-12] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is down
2021-12-07 16:05:05.634  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-4] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is up
2021-12-07 16:05:05.635  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-9] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:05:05.844  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-9] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:05:22.662  WARN [ocp-permissions,,,] 1 --- [sson-netty-2-14] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is down
2021-12-07 16:05:23.663  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-8] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is up
2021-12-07 16:05:23.665  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-13] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:05:23.667  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-14] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:06:38.777  WARN [ocp-permissions,,,] 1 --- [isson-netty-2-2] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is down
2021-12-07 16:06:39.779  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-11] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is up
2021-12-07 16:06:39.780  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-7] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:06:39.782  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-1] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:07:19.843  WARN [ocp-permissions,,,] 1 --- [isson-netty-2-6] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is down
2021-12-07 16:07:20.846  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-13] o.r.c.SentinelConnectionManager          : slave: redis://10.131.3.187:6379 is up
2021-12-07 16:07:20.849  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-1] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:07:20.850  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-2] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.131.3.187/10.131.3.187:6379
2021-12-07 16:07:38.874  WARN [ocp-permissions,,,] 1 --- [isson-netty-2-1] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is down
2021-12-07 16:07:39.876  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-8] o.r.c.SentinelConnectionManager          : slave: redis://10.130.5.238:6379 is up
2021-12-07 16:07:39.878  INFO [ocp-permissions,,,] 1 --- [sson-netty-2-14] o.r.c.pool.PubSubConnectionPool          : 1 connections initialized for 10.130.5.238/10.130.5.238:6379
2021-12-07 16:07:39.880  INFO [ocp-permissions,,,] 1 --- [isson-netty-2-1] o.r.connection.pool.SlaveConnectionPool  : 24 connections initialized for 10.130.5.238/10.130.5.238:6379
```

Please kindly note that the above behaviour in application logs could not be reproduced with bitnami/redis helm chart 14.5.8. The difference is that binami/redis helm chart 15.5.5 has enabled DNS/hostnames (announce-hostnames). 

Could you, please, elaborate whether enabling DNS/hostnames is related to the issue above?

Thanks,
Yannis

**Comment by mrniko** [comment]
@gkormaris 

it's related to a wrong configuration of sentinel servers. They should return same hosts.

**Comment by gkormaris** [comment]
I use the sentinelServersConfig YAML configuration below.

```
sentinelServersConfig:
  idleConnectionTimeout: 10000
  connectTimeout: 3000
  timeout: 3000
  retryAttempts: 3
  retryInterval: 1500
  failedSlaveReconnectionInterval: 3000
  failedSlaveCheckInterval: 60000
  password: ********
  subscriptionsPerConnection: 5
  loadBalancer: !<org.redisson.connection.balancer.RoundRobinLoadBalancer> {}
  subscriptionConnectionMinimumIdleSize: 1
  subscriptionConnectionPoolSize: 20
  slaveConnectionMinimumIdleSize: 24
  slaveConnectionPoolSize: 128
  masterConnectionMinimumIdleSize: 24
  masterConnectionPoolSize: 128
  readMode: "SLAVE"
  subscriptionMode: "SLAVE"
  pingConnectionInterval: 2000
  sentinelAddresses:
    - "redis://my-redis-headless.playground.svc.cluster.local:26379"
  masterName: "my-redis"
  database: 0
threads: 16
nettyThreads: 32
codec: !<org.redisson.codec.JsonJacksonCodec> {}
transportMode: "NIO"
``` 
I have the Services below which are created by bitnami/redis chart.

```
# kubectl get svc | grep my-redis
my-redis                 ClusterIP   172.30.93.189    <none>        6379/TCP,26379/TCP 
my-redis-headless        ClusterIP   None             <none>        6379/TCP,26379/TCP 
my-redis-metrics         ClusterIP   172.30.105.60    <none>        9121/TCP           
```

I have executed the command `sentinel slaves my-redis` in all my sentinels and everything seems good from a Redis Server/Sentinel point of view.

The issue remains in my SpringBoot Application.

**Comment by gkormaris** [comment]
This issue is not reproducible when nettyThreads is set to 1 which makes me think that this is some kind of thread safety/race condition issue.

I debugged the code and it is the latch call below inside checkSlavesChange which logs the aforementioned log entries.

```
            latch.latch(() -> {
                MasterSlaveEntry entry = getEntry(singleSlotRange.getStartSlot());
                entry.getAllEntries().stream()
                        .map(e -> e.getClient().getAddr())
                        .map(a -> toURI(a.getAddress().getHostAddress(), String.valueOf(a.getPort())))
                        .filter(a -> !currentSlaves.contains(a) && !a.equals(currentMaster.get()))
                        .forEach(a -> slaveDown(a));
            }, slavesMap.size());
```

In my opinion the currentSlaves HashSet could be accessed by more than one threads and it is not thread safe since it is defined as below.

`Set<RedisURI> currentSlaves = new HashSet<>(slavesMap.size());
`
@mrniko Could you, please, check if this is a possible bug please?

**Comment by mrniko** [comment]
Fixed. Thanks for report

**Comment by aeimer** [comment]
Hi @mrniko @gkormaris 

We have the version `16.4.0` of the bitnami helm chart installed and the redison version `redison-spring-data-26@3.16.8` is still producing these logs:
![image](https://user-images.githubusercontent.com/25035182/154449104-4c8c8b60-1033-443f-ab7c-9dfc2ddf4996.png)

We have an open issue at bitnami with more extensive logs: https://github.com/bitnami/charts/issues/8614

**Comment by mrniko** [comment]
@aeimer 

Did you try to set `nettyThreads = 1`  ?

**Comment by tschroe** [comment]
Hi @mrniko 

We have tried it out and with `nettyThreads=1` the issue doesn't appear. So, could it be that it's the same bug again?

(I'm in the same team as @aeimer)

**Comment by mrniko** [comment]
@tschroe 

Please make sure that Redisson 3.16.7+ is in classpath.

**Comment by aeimer** [comment]
@mrniko We actually recreated a virgin redis helm deployment (even removed the PVC!) then it actually worked.
Thanks for the help though!

**Comment by jaudiger** [comment]
Hi @mrniko,

I'm facing the exact same issue, see the logs attached below:

```json
{"@timestamp":"2022-03-18T16:34:19.239Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:25.25Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:25.253Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:25.255Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:26.253Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:28.258Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:28.26Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:28.263Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:29.26Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:40.283Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:40.285Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:40.287Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:41.285Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:42.288Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:42.29Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:42.292Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:44.293Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:51.305Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:51.307Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:51.309Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:52.309Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:34:58.319Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:34:58.321Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:58.323Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:34:59.32Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:35:02.327Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:35:02.33Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:02.331Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:05.333Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:35:22.359Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:35:22.361Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:22.364Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:23.361Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:35:27.37Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:35:27.372Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:27.374Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:29.373Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
{"@timestamp":"2022-03-18T16:35:33.381Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"slave: redis://10.42.0.180:6379 is up"}
{"@timestamp":"2022-03-18T16:35:33.387Z","@version":"1","logger_name":"org.redisson.connection.pool.PubSubConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"1 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:33.394Z","@version":"1","logger_name":"org.redisson.connection.pool.SlaveConnectionPool","thread_name":"redisson-netty-2-1","level":"INFO","level_value":20000,"message":"24 connections initialized for 10.42.0.180/10.42.0.180:6379"}
{"@timestamp":"2022-03-18T16:35:34.387Z","@version":"1","logger_name":"org.redisson.connection.SentinelConnectionManager","thread_name":"redisson-netty-2-1","level":"WARN","level_value":30000,"message":"slave: redis://10.42.0.180:6379 is down"}
```

I'm currently using the bitnami chart to deploy Redis sentinels with the following configuration:

```yaml
sentinel:
  enabled: true
  masterSet: redis-master
  downAfterMilliseconds: 16000
  failoverTimeout: 8000
  quorum: 2
  parallelSyncs: 1

replica:
  replicaCount: 3
```

On the service side, I'm using the latest version of Redisson available which is `3.16.8`. I even tested with a single Netty thread, and the issue is still here.

Maybe the issue comes from Redis sentinels, but from what I tested with the CLI, I didn't find any relevant issue:

```bash
➜ k exec -it redis-node-0 -c redis -- bash -c "redis-cli -h localhost -p 26379 sentinel get-master-addr-by-name redis-master | head -n 1"
redis-node-2.redis-headless.default.svc.cluster.local

➜ k exec -it redis-node-1 -c redis -- bash -c "redis-cli -h localhost -p 26379 sentinel get-master-addr-by-name redis-master | head -n 1"
redis-node-2.redis-headless.default.svc.cluster.local

➜ k exec -it redis-node-2 -c redis -- bash -c "redis-cli -h localhost -p 26379 sentinel get-master-addr-by-name redis-master | head -n 1"
redis-node-2.redis-headless.default.svc.cluster.local

➜ k exec -it redis-node-2 -c redis -- redis-cli -h redis-node-2.redis-headless.default.svc.cluster.local
redis-node-2.redis-headless.default.svc.cluster.local:6379> info replication
# Replication
role:master
connected_slaves:2
slave0:ip=redis-node-1.redis-headless.default.svc.cluster.local,port=6379,state=online,offset=13681281,lag=0
slave1:ip=redis-node-0.redis-headless.default.svc.cluster.local,port=6379,state=online,offset=13681281,lag=0
master_failover_state:no-failover
master_replid:d7de8e84b2be002d0d5e5b01f5429eb7a79d59fb
master_replid2:dd55756fabc189fcd8c06193b46ee1e832b52fe9
master_repl_offset:13681281
second_repl_offset:13415052
repl_backlog_active:1
repl_backlog_size:1048576
repl_backlog_first_byte_offset:13387028
repl_backlog_histlen:294254
```
